esp32/mods/modbt.c

```diff
@@ -1445,35 +1415,47 @@ static mp_obj_t modbt_connect(mp_obj_t addr)
 
 STATIC mp_obj_t bt_set_advertisement_params (mp_uint_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
```

Comment:
The problem with original code is that if we call `bt_set_advertisement_params()` with keyword argument(s), than unmentioned `bt_adv_params` will be reinitiated, which is probably not the intended behaviour of the function.

Example with **original** code:
1. Calling `bt_set_advertisement_params(adv_int_max = 128)` 
      - `bt_adv_params.adv_int_min = 32`
      - `bt_adv_params.adv_int_max = 128`
1. Calling `bt_set_advertisement_params(adv_int_min = 16)` 
      - `bt_adv_params.adv_int_min = 16`
      - **`bt_adv_params.adv_int_max = 64`** <- adv_int_max is reinitialized

Example with **new** code:
1. Calling `bt_set_advertisement_params(adv_int_max = 128)` 
      - `bt_adv_params.adv_int_min = 32`
      - `bt_adv_params.adv_int_max = 128`
1. Calling `bt_set_advertisement_params(adv_int_min = 16)` 
      - `bt_adv_params.adv_int_min = 16`
      - **`bt_adv_params.adv_int_max = 128`** <- adv_int_max remains the same

The behaviour is tested on LoPy4, I suggest to accept all the modifications inside `bt_set_advertisement_params()` function definition.